python/pyarrow/tests/test_plasma.py

```diff
@@ -390,6 +390,12 @@ def __init__(self, val):
             object_id, -1, serialization_context=serialization_context)
         assert result.val == val.val
 
+    def test_store_np_matrix(self):
```

Comment:
Can we test a few more dtypes as well?

Reply:
I've added other types. Do you think `str`, `int` and `float` is enough?

Reply:
That's pretty good, but are we actually hitting the `if obj.dtype.str != '|O':` code path? Maybe add a test that uses a numpy matrix that contains some custom object, like

```python
class Foo(object):
    pass

x = np.matrix([Foo()])
```

There should be an analogous test for numpy arrays. If so, maybe copy that one.

Reply:
I've added `CustomType` to the list of types we check for. I've added some logging locally and we appear to be hitting `if obj.dtype.str != '|O':` now.
We do get a lot of `np.matrix` deprecation warnings, should we silence them?

Reply:
> We do get a lot of np.matrix deprecation warnings, should we silence them?

In tests you mean?

Reply:
Yes tests trigger them. I am not sure how it works if this is merged into arrow - would every deserialization trigger warnings or not?

Reply:
I think we could silence them during tests. But during regular use I would just leave to the user to do so if they want.

Reply:
I don't have a strong opinion either way. What is the project default @robertnishihara?

Reply:
I don't see the warnings. Where are they? Did they appear in the arrow CI?

Reply:
Odd - locally I see a warning similar to [this one](https://travis-ci.org/apache/arrow/jobs/519253981#L3149) about `np.matrix` being deprecated. I can't find  it in the CI log. I like @mitar's point I'll leave this as is.